src/modules/dataProviders/providers/squid/squidSubscription.provider.ts

```diff
@@ -0,0 +1,218 @@
+    subscriptionEntitiesList: Array<
+      | SquidSubscriptionNotificationsResponseDto
+      | SquidSubscriptionsActivitiesResponseDto
+    >
+  ) {
+    const subscriptionStatusData =
+      await this.dataProvidersService.getOrCreateStatusByQueryName({
+        name: subQueryName
+      });
+
+    switch (subQueryName) {
+      case SquidApiQueryName.activities:
+        return (<Array<SquidActivitiesResponseDto>>(
+          subscriptionEntitiesList
+        )).filter(
+          (activity) =>
+            Number.parseInt(activity.blockNumber) >
+            subscriptionStatusData.lastProcessedBlockNumber
+        );
+        break;
+      case SquidApiQueryName.notifications:
+        return (<Array<SquidNotificationsResponseDto>>(
+          subscriptionEntitiesList
+        )).filter(
+          (notification) =>
+            Number.parseInt(notification.activity.blockNumber) >
+            subscriptionStatusData.lastProcessedBlockNumber
+        );
+        break;
+      default:
+    }
+  }
+
+  async handleNewSubDataNotifications(
+    notifications: Array<SquidNotificationsResponseDto>
+  ) {
+    for (const squidNotification of notifications) {
+      const { activity, account, id } = squidNotification;
+      switch (squidNotification.activity.event) {
+        case EventName.CommentReplyCreated:
+          await this.notificationService.handleNotificationEventForSubstrateAccount(
+            {
+              eventName: activity.event,
+              substrateAccountId: account.id,
+              post: activity.post
+            }
+          );
+          break;
+
+        case EventName.ExtensionDonationCreated:
+          await this.notificationService.handleNotificationEventForSubstrateAccount(
+            {
+              eventName: activity.event,
+              substrateAccountId: account.id,
+              post: activity.post,
+              extension: activity.extension
+            }
+          );
+          break;
+
+        default:
+      }
+    }
+  }
+
+  /**
+   * Removes Notifications from received list if it's "PostCreated || CommentCreated || CommentReplyCreated" event and
+   * Post entity contains ContentExtension, which has its own Squid event (subsocial_donations || subsocial_evm_nft).
+   * Such Post is sort of wrapper/container for ContentExtension and its creation event is redundant in context of
+   * Notifications service.
+   *
+   * @param notifications
+   */
+  filterSubDataNotificationsByContentExtensionWrappers(
+    notifications: Array<SquidNotificationsResponseDto>
+  ) {
+    const filteredNotifications: Array<SquidNotificationsResponseDto> = [];
+
+    for (const notificationData of notifications) {
+      if (
+        (notificationData.activity.event !== EventName.PostCreated &&
+          notificationData.activity.event !== EventName.CommentCreated &&
+          notificationData.activity.event !== EventName.CommentReplyCreated) ||
+        !notificationData.activity.post
```

Comment:
Why we are doing the sorting here? Can we not use filter in query?

Reply:
Because we need all data here. This loop removes notifications about `wrapper` events. It means that we don't need notify user about `reply` & `donation`. So `reply` as Post is a wrapper entity for Donation Extension. In Squid we have 2 separate events `CommentReplyCreated` and `ExtensionDonationCreated`.